wallet/wallet_test.go

```diff
@@ -210,17 +210,56 @@ func TestWallet_CreateTicket(t *testing.T) {
 	require.NoError(t, err)
 
 	t.Run("Returns real ticket and nil error with good params", func(t *testing.T) {
-		proof := types.PoStProof{0xbb}
-		ticket, err := consensus.CreateTicket(proof, addr, w)
+		parent := types.Ticket{VDFResult: types.VDFY{0xbb}}
+		ticket, err := consensus.TicketMachine{}.NextTicket(parent, addr, w, 0)
 		assert.NoError(t, err)
 		assert.NotNil(t, ticket.VRFProof)
+		assert.NoError(t, consensus.TicketMachine{}.NotarizeTime(&ticket))
+
+		// wallet can validate signature
+		v, err := consensus.TicketMachine{}.ValidateTicket(parent, ticket, addr, 0)
+		assert.NoError(t, err)
+		assert.True(t, v)
 	})
 
 	t.Run("Returns error and empty ticket when signer is invalid", func(t *testing.T) {
-		proof := types.PoStProof{0xc0}
+		parent := types.Ticket{VDFResult: types.VDFY{0xc0}}
 		badAddress := address.TestAddress
-		ticket, err := consensus.CreateTicket(proof, badAddress, w)
+		ticket, err := consensus.TicketMachine{}.NextTicket(parent, badAddress, w, 0)
 		assert.Error(t, err, "t, SignBytes error in CreateTicket: public key not found")
 		assert.Equal(t, types.VRFPi(nil), ticket.VRFProof)
 	})
 }
+
+func TestWalletRunElection(t *testing.T) {
+	tf.UnitTest(t)
+
+	ds := datastore.NewMapDatastore()
+	fs, err := wallet.NewDSBackend(ds)
+	assert.NoError(t, err)
+	w := wallet.New(fs)
+	addr, err := wallet.NewAddress(w)
+	require.NoError(t, err)
+
+	t.Run("Returns real election proof and nil error with good params", func(t *testing.T) {
+		parent := types.Ticket{VDFResult: types.VDFY{0xbb}}
+		ep, err := consensus.ElectionMachine{}.RunElection(parent, addr, w)
+		assert.NoError(t, err)
+		assert.NotNil(t, ep)
+
+		// wallet can validate signature
+		pk, err := w.GetPubKeyForAddress(addr)
+		require.NoError(t, err)
+		v, err := w.Verify(parent.VDFResult[:], pk, types.Signature(ep[:]))
+		assert.NoError(t, err)
+		assert.True(t, v)
+	})
+
+	t.Run("Returns error and empty election proof when signer is invald", func(t *testing.T) {
+		parent := types.Ticket{VDFResult: types.VDFY{0xbb}}
+		badAddress := address.TestAddress
+		ep, err := consensus.ElectionMachine{}.RunElection(parent, badAddress, w)
+		assert.Error(t, err)
+		assert.Equal(t, types.VRFPi{}, ep)
+	})
+}
```

Comment:
These tests do not seem to have much to do with the wallet at all. Are they just testing it can sign things? We should just test that the `Sign` method works here, and move the rest of the test to `consensus` package (with a 
minimal signer).